### DISCUSSION TITLE
Sonic CD-Esque Spin Dash

### DISCUSSION THREAD

**Discussion by July2612** [TrianglyRU/OrbinautFramework#27] [Show and tell]
Since this _is_ the show and tell section, I show you my attempt at recreating the Sonic CD spin dash in Orbinaut 2! 

https://github.com/user-attachments/assets/6f90a4e2-8441-4e19-86c6-4296084cfa22

In my private fork of Orbinaut 2 this is something I can tweak in `scr_fw_game_setup` like the other movesets, using `SPIN_GEN` and `SPIN_CD` as the usable options. I could have just used 0 and 1 like the rotation options, but I like the macros, they're cool!

![image](https://github.com/user-attachments/assets/64c5dbee-c0ca-45d6-bae7-ae9789df9bae)

I might do a video and/or text tutorial for this later down the road if someone is interested. Maybe I'll even show _two_ methods of doing so:

- Replacing the default spin dash
- Adding it _alongside_ the default spin dash (making it a changeable global variable)

The latter is probably preferable to most people, but if you're someone out there who wants to make a fully CD-style fangame inside of Orbinaut, the former option may be the one you need!

As for the basics of how I did it, I modified the peel out script and tossed it into the spin dash script. Obvious changes aside, I also removed the peel out's requirement where you have to charge it all the way to use it. This is so you can do weaker spin dashes, if you need to. 

I also had to make a new animation slot for the CD spin dash, since using the default roll animation caused you to begin sliding down slopes once you started to charge, amongst other things.

I apologize for the amount of words above. I like explaining the thoughts that roam in my head. Anyways, that's all for now. If I figure out how to add anything else (the strike dash from the game gear games comes to mind as something I'd love to add one day 👀), I'll make sure to do another post talking about it! Later!

**Comment by TrianglyRU** [comment]
Good stuff! 👏

**Comment by RunninPigeon** [comment]
I would love a tutorial about this!

**Comment by Ulises2020-X** [comment]
Make a tutorial please!
